packages/appium/lib/utils.js

```diff
@@ -34,18 +33,42 @@ function inspectObject (args) {
   }
 }
 
-function parseExtensionArgs (extensionArgs, extensionName) {
-  if (!_.isString(extensionArgs)) {
+function getExtensionArgs (extensionArgs, extensionName) {
+  if (!_.has(extensionArgs, extensionName)) {
     return {};
   }
-  const parsedExtensionArgs = parseJsonStringOrFile(extensionArgs);
-  const extensionSpecificArgs = parsedExtensionArgs[extensionName];
-  if (!_.isPlainObject(extensionSpecificArgs)) {
+  if (!_.isPlainObject(extensionArgs[extensionName])) {
     throw new Error(`Driver or plugin arguments must be plain objects`);
   }
-  return extensionSpecificArgs;
+  return extensionArgs[extensionName];
 }
 
+
+function ensureNoUnknownArgs (extensionArgs, argsConstraints) {
+  const knownArgNames = Object.keys(argsConstraints);
+  for (const argName of Object.keys(extensionArgs)) {
+    if (!knownArgNames.includes(argName)) {
```

Comment:
I would rather collect all unsupported values first. We could quickly this by calling `_.difference` lodash helper